rust/xaynet-core/src/mask/object/mod.rs

```diff
@@ -103,66 +100,55 @@ impl MaskOne {
         }
     }
 
-    pub fn empty(config: MaskConfig) -> Self {
+    /// Creates a new mask unit of given masking configuration with default value `1`.
+    pub fn default(config: MaskConfig) -> Self {
         Self {
-            data: BigUint::from(1_u8), // NOTE not really empty!
+            data: BigUint::from(1_u8),
             config,
         }
     }
 
-    /// Checks if this mask object conforms to the given masking configuration.
+    /// Checks if the data value conforms to the masking configuration.
     pub fn is_valid(&self) -> bool {
         self.data < self.config.order()
     }
 }
 
 #[derive(Debug, Hash, PartialEq, Eq, Clone, Serialize, Deserialize)]
-/// A mask object wrapper around a `MaskMany`, `MaskOne` pair.
+/// A mask object consisting of a vector part and unit part.
 pub struct MaskObject {
-    pub vector: MaskMany,
-    pub scalar: MaskOne,
+    pub vect: MaskVect,
+    pub unit: MaskUnit,
 }
 
 impl MaskObject {
-    // TODO doc
-    pub fn new(vector: MaskMany, scalar: MaskOne) -> Self {
-        Self { vector, scalar }
+    /// Creates a new mask object from the given vector and unit.
+    pub fn new(vect: MaskVect, unit: MaskUnit) -> Self {
+        Self { vect, unit }
     }
 
-    // TODO perhaps no need
-    pub fn new_unchecked(
-        config_v: MaskConfig,
-        data_v: Vec<BigUint>,
-        config_s: MaskConfig,
-        data_s: BigUint,
-    ) -> Self {
-        Self {
-            vector: MaskMany::new(config_v, data_v),
-            scalar: MaskOne::new(config_s, data_s),
-        }
-    }
-
-    // TODO doc
+    /// Creates a new mask object from the given vector, unit and masking configurations.
     pub fn new_checked(
```

Comment:
I'm not sure we need this method. Could we have only `MaskObject::new` which takes a `MaskVect` and `MaskUnit` that are assumed to be valid, and `Mask{Vect,Unit}::new` that always perform the validation? That way we'd also remove `Mask{Vect,Unit}::new_checked`.

Also, if we really need both the checked and unchecked variant, my personal perference would be to have `new` and `new_unchecked` instead of `new_checked` and `new`. It is more obvious for the caller that the method skips some potentially important checks.

Reply:
agree, the `..._unchecked` notation is also consistent with the std library.

Reply:
thanks both. Renamed `new`/`new_checked` to `new_unchecked`/`new` accordingly. Currently there's use of both so I've left both variants in, but I agree it can probably be consolidated, at a later date.